### DISCUSSION TITLE
Generators

### DISCUSSION THREAD

**Discussion by iYadgar** [e-square-io/angular-cli-gui#46] [High Level Design]
### Overview and Goals
Create generators view, that will hold all angular’s cli generate commands, for POC we’ll start with the most common ones. each generator will hold a route: 

- generators/component
- generators/directive
- generators/service
- generators/module

There will be a dynamic form that will hold the inputs according the parameters sent from the server for each generator.

On every form change event we’ll run a ‘dry run’ command for the generator to show the live output, for POC we’ll not implement socket but we’ll send an http request for each form change.

**There will a be checkbox to opt in/out the ‘dry run’ feature and ‘Run Task’ to actually run the command.

The output will be presented with a “CLI” like component.

---

### Functional Requirements 
The main requirement for the feature to be feature ready is to be able run the commands according the parameters supplied by the user and present the actual results.

---

### Architecture and Design
![Untitled drawio (1) (1)](https://user-images.githubusercontent.com/63288982/220120459-436c8b1e-8b2b-4eda-b8c7-3747a0a89d98.png)

---

### Data Model
```
interface Generator<T> {
	name:string;
	formDetails?: FormDetail<T>[];
}
interface FormDetail<T>{
	params: T,
...whatever interface it will be on the form library we use
}
```
---

### User Interface Design

![Untitled](https://user-images.githubusercontent.com/63288982/220120410-bcb32ccc-baf1-4c0d-aeb2-9e81d9293133.png)

**Comment by ronnetzer** [comment]
I think some HLD should be done for the BE side as well: how do you plan to "understand" the configuration of each schematic? also is there gonna be some kind of "discovery" process? like going over the packages installed in a workspace, figuring out which packages has schematics in them and return a list of all available schematics
